packages/dds/tree/src/feature-libraries/schemaIndexFormat.ts

```diff
@@ -158,6 +223,8 @@ function decodeTree(schema: TreeSchemaFormat): TreeSchema {
  */
 export function getSchemaString(data: SchemaData): string {
 	const encoded = encodeRepo(data);
+	assert(Versioned.Check(encoded), "Encoded schema should be versioned");
```

Comment:
was the intent here to just get extra specific validation for versioning? Or some sort of forward-compat-thinking check? `CompiledFormat` type includes a version check, though it's for a literal vs. "some string"

Reply:
We want to ensure our data is compatible with version detection, as well as the current version to catch cases where the format changes break version detection. Its not likely, but the check was trivial to include and had nice symmetry with the validation we do when parsing.